examples/tutorial/TUTORIAL.md

```diff
@@ -0,0 +1,191 @@
+
+The gRPC service we will be implementing provides two amazing functions:
+
+1. `Add`, which adds two integers.
+2. `RunningSum`, which receives a stream of integers from the client and finally returns a single integer that is the sum of all the integers it has received.
+
+You can run the examples by running `stack exec arithmetic-server` and `stack exec arithmetic-client`.
+
+### Library Organization
+
+**tl;dr: you probably only need to import `Network.GRPC.HighLevel.Generated`.** Other modules are exposed for advanced users only.
+
+This library exposes quite a few modules, but you won't need to worry about most of them. They are currently organized based on the level of abstraction they afford over using the C [gRPC Core library](http://www.grpc.io/grpc/core/) directly:
+
+* *`Unsafe`* modules directly wrap functions in the gRPC Core library. Using them directly is like using C: you need to think about memory management, pointers, and so on. The rest of the library is built on top of these functions and users of gRPC-haskell should never need to deal with the `Unsafe` modules directly.
+* *`LowLevel`* modules still require an understanding of the gRPC Core library, but guarantee memory and thread safety. Only advanced users with special requirements would use `LowLevel` modules directly.
+* *`HighLevel`* modules give you an opinionated Haskell interface to gRPC that should cover most use cases while (hopefully) being easy to use. You should only need to import the `Network.GRPC.HighLevel.Generated` module to start using the library. If you need to import other modules, we probably forgot to re-export something and you should open an issue or PR.
+
+### Getting started
+
+To start out, we need to generate code for our protocol buffers and RPCs.
+
+```
+stack exec -- compile-proto-file --proto examples/echo/echo.proto > examples/echo/echo-hs/Echo.hs
+```
+
+The `.proto` file compiler always names the generated module the same as the `.proto` file, capitalizing the first letter if it is not already. Since our proto file is `arithmetic.proto`, the generated code should be placed in `Arithmetic.hs`.
+
+The important things to notice in this generated file are:
+
+1. For each proto message type, an equivalent Haskell type with the same name has been generated.
+2. The `arithmeticServer` function takes a a record containing handlers for each RPC endpoint and some options, and starts a server. So, you just need to call this function to get a server running.
+3, The `arithmeticClient` function takes a `Client` (which is just a proof that the gRPC core has been started) and gives you a record of functions that can be used to run RPCs.
+
+### The server
+
+First, we need to turn on some language extensions:
+
+```
```

Comment:
Minor suggestion, in most of these Haskell code-blocks I think you should be able to do this:

```
```haskell
```

and it will syntax high-light the code for Haskell.